digiroad2-geo/src/main/scala/fi/liikennevirasto/digiroad2/lane/LanePartitioner.scala

```diff
@@ -133,9 +133,25 @@ object LanePartitioner {
   // be geometrically connected together)
   def partition(allLanes: Seq[PieceWiseLane], roadLinks: Map[Long, RoadLink]): Seq[Seq[PieceWiseLane]] = {
 
+    def groupAdditionalLanes(mainLaneGroups: Seq[Seq[PieceWiseLane]], allAdditionalLanes: Seq[PieceWiseLane]): Seq[Seq[PieceWiseLane]] = {
+      mainLaneGroups.flatMap(mainLaneGroup => {
+        val mainLaneLinkIdsAndSideCodes = mainLaneGroup.map(lane => (lane.linkId, lane.sideCode))
+        val additionalLanesOnLinks = allAdditionalLanes.filter(lane => mainLaneLinkIdsAndSideCodes.map(_._1).contains(lane.linkId))
+        val lanesWithCorrectSideCode = additionalLanesOnLinks.filter(lane => {
+          val linkIdAndSideCode = (lane.linkId, lane.sideCode)
+          mainLaneLinkIdsAndSideCodes.contains(linkIdAndSideCode)
+        })
+        val additionalLanesGroupedByLaneCode = lanesWithCorrectSideCode.groupBy(additionalLane => {
+          val laneCode = additionalLane.laneAttributes.find(_.publicId == "lane_code")
+          laneCode
+        })
+        additionalLanesGroupedByLaneCode.values.toSeq
+      })
+    }
+
     //Groups lanes by roadIdentifier, sideCode, LaneCode, and other lanes on link.
     //Makes sure that all the lanes in group are connected and there are no gaps in lane selection
-    def groupLanes(lanes: Seq[PieceWiseLane]): Seq[Seq[PieceWiseLane]] = {
+    def groupMainLanes(lanes: Seq[PieceWiseLane]): Seq[Seq[PieceWiseLane]] = {
```

Comment:
tää toimii nyt vähän oudosti metodi sanoo groupMainLanes mutta groupMainLanes on funktion viimeisiä operaatioita

Reply:
missä partitionedByAdditional käytetään? En nyt löydä sille mitään käyttöä

Reply:
Okey tätä koko partition hommaa alkaa olla aika vaikeeta lukea

Reply:
partitionedByAdditional käytetään seuraavassa kohdassa rivillä 180 kun näille joukoille haetaan viereiset kaistat, jotta voidaan selvittää mitkä joukot ovat oikeasti jatkuvia. Tämä sen takia, että voidaan varmistaa, että valitut kaistat ovat vierekkäin, eikä väliin ole jäänyt aukkoja

Reply:
Myönnän, että tää lanePartitioner on turhan monimutkanen, mutta helpompaa tapaa muodostaa haluttuja kaistajoukkoja ei taida olla, tai ainakaan itse en ole keksinyt

Reply:
joo löysin lopulta mistä homma alkoi ja loppui, jotakin laajempaa huoltoo kyl vaatisi että saisi jotain selkoa mitä tehtdään

Reply:
ainkin noi groupMainLane ja groupAdditionalLanes vois ehkä heittää ulos tosta metodista